explorer/explorermiddleware.go

```diff
@@ -66,6 +66,22 @@ func (exp *explorerUI) BlockHashPathOrIndexCtx(next http.Handler) http.Handler {
 	})
 }
 
+// SyncStatusPageActivation serves only the syncing status page until its
+// deactivated when DisplaySyncStatusPage is set to false. This page is served
+// for all the possible routes supported until the background syncing is done.
+func (exp *explorerUI) SyncStatusPageActivation(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if exp.DisplaySyncStatusPage() {
+			exp.StatusPage(w, "Database Update Running. Please Wait...",
+				"Blockchain sync is running. Please wait ...", BlockchainSyncingType)
```

Comment:
In testing, I discovered that `BlockchainSyncingType` as the `statusType` results in `http.StatusServiceUnavailable` (503), however I'm not certain we should return an error code.  I think that 503 Service Unavailable does make sense though:

> The server is currently unable to handle the request due to a temporary overloading or maintenance of the 
server. The implication is that this is a temporary condition which will be alleviated after some delay. If known, the length of the delay MAY be indicated in a Retry-After header. If no Retry-After is given, the client SHOULD handle the response as it would for a 500 response.

My only concern here is how a client would handle the response.  We have seen websocket connections broken when pages reload with turbolinks if the initial response code was not a 200.  @SeaLightHouse ran into this.

We might consider 202: https://httpstatuses.com/202

> The request has been accepted for processing, but the processing has not been completed. The request might or might not eventually be acted upon, as it might be disallowed when processing actually takes place.

> There is no facility in HTTP for re-sending a status code from an asynchronous operation.

> The 202 response is intentionally noncommittal. Its purpose is to allow a server to accept a request for some other process (perhaps a batch-oriented process that is only run once per day) without requiring that the user agent's connection to the server persist until the process is completed. **The representation sent with this response ought to describe the request's current status and point to (or embed) a status monitor that can provide the user with an estimate of when the request will be fulfilled.**

Reply:
Honestly I haven't given much thought to the http status code 503. When the user tries to access a page that is restricted while the sync in progress and the server doesn't serve it, we should not classify that as an error with 503 usage but 202 makes more sense in this case scenario.